File: Frontend/pharmind-web/src/config/env.ts
/**
 * Configuración de variables de entorno
 * Todas las variables de entorno deben comenzar con VITE_ para ser accesibles en el cliente
 */

export const env = {
  // API Configuration
  apiBaseUrl: import.meta.env.VITE_API_BASE_URL || 'http://localhost:5209/api',

  // Application Configuration
  appName: import.meta.env.VITE_APP_NAME || 'PharMind',
  environment: import.meta.env.VITE_ENVIRONMENT || 'development',

  // Feature Flags
  enableDebugMode: import.meta.env.VITE_ENABLE_DEBUG_MODE === 'true',
  enableAnalytics: import.meta.env.VITE_ENABLE_ANALYTICS === 'true',

  // SignalR
  signalRHubUrl: import.meta.env.VITE_SIGNALR_HUB_URL || import.meta.env.VITE_API_BASE_URL?.replace('/api', '') || 'http://localhost:5209',

  // Computed properties
  isDevelopment: import.meta.env.MODE === 'development',
  isProduction: import.meta.env.MODE === 'production',
  isStaging: import.meta.env.VITE_ENVIRONMENT === 'staging',
} as const;

// Validar configuración crítica en desarrollo
if (env.isDevelopment && !env.apiBaseUrl) {
  console.warn('⚠️ VITE_API_BASE_URL no está configurado. Usando valor por defecto.');
}

// Exportar para facilitar debugging
if (env.enableDebugMode) {
  console.log('🔧 Environment Configuration:', {
    apiBaseUrl: env.apiBaseUrl,
    environment: env.environment,
    mode: import.meta.env.MODE,
  });
}

export default env;

File: Mobile/pharmind_mobile/lib/config/env.dart
/// Configuración de variables de entorno para PharMind Mobile
/// Usa --dart-define para configurar en tiempo de compilación
class AppConfig {
  // API Configuration
  static const String apiBaseUrl = String.fromEnvironment(
    'API_BASE_URL',
    defaultValue: 'http://10.0.2.2:5209/api', // Emulador Android
  );

  // Application Configuration
  static const String appName = String.fromEnvironment(
    'APP_NAME',
    defaultValue: 'PharMind',
  );

  static const String environment = String.fromEnvironment(
    'ENVIRONMENT',
    defaultValue: 'development',
  );

  // Feature Flags
  static const bool enableDebugMode = bool.fromEnvironment(
    'ENABLE_DEBUG_MODE',
    defaultValue: true,
  );

  static const bool enableOfflineMode = bool.fromEnvironment(
    'ENABLE_OFFLINE_MODE',
    defaultValue: true,
  );

  // Timeouts (en segundos)
  static const int connectTimeout = int.fromEnvironment(
    'CONNECT_TIMEOUT',
    defaultValue: 30,
  );

  static const int receiveTimeout = int.fromEnvironment(
    'RECEIVE_TIMEOUT',
    defaultValue: 30,
  );

  // Computed properties
  static bool get isDevelopment => environment == 'development';
  static bool get isProduction => environment == 'production';
  static bool get isStaging => environment == 'staging';

  /// Imprime la configuración actual (solo en modo debug)
  static void printConfig() {
    if (enableDebugMode) {
      // ignore: avoid_print
      print('🔧 App Configuration:');
      // ignore: avoid_print
      print('   Environment: $environment');
      // ignore: avoid_print
      print('   API Base URL: $apiBaseUrl');
      // ignore: avoid_print
      print('   App Name: $appName');
      // ignore: avoid_print
      print('   Debug Mode: $enableDebugMode');
      // ignore: avoid_print
      print('   Offline Mode: $enableOfflineMode');
      // ignore: avoid_print
      print('   Connect Timeout: ${connectTimeout}s');
      // ignore: avoid_print
      print('   Receive Timeout: ${receiveTimeout}s');
    }
  }

  /// Valida que la configuración esté correcta
  static void validate() {
    if (apiBaseUrl.isEmpty) {
      throw Exception('API_BASE_URL no puede estar vacía');
    }

    if (isProduction && apiBaseUrl.contains('localhost')) {
      throw Exception(
        'PRODUCCIÓN no puede usar localhost. Configure API_BASE_URL correctamente.',
      );
    }

    if (isProduction && apiBaseUrl.contains('10.0.2.2')) {
      throw Exception(
        'PRODUCCIÓN no puede usar IP del emulador. Configure API_BASE_URL correctamente.',
      );
    }
  }
}
